src/KOKKOS/pair_eam_kokkos.cpp

```diff
@@ -31,9 +31,25 @@
 #include "pair_kokkos.h"
 
 #include <cmath>
-
 using namespace LAMMPS_NS;
 
+#define MAX_CACHE_ROWS 500
+#define MAX_CACHE_COLS 7
+
+/* ---------------------------------------------------------------------- */
+template<class DeviceType, class tag>
+auto policyInstance(int inum){
```

Comment:
This is a really nice design, I like it. I think it might make more sense as a functor member function so you can query info from the functor --- for example, instead of having `MAX_CACHE_ROWS` you can query the compile-time `7` in the size of the spline structure.

Reply:
This is a good idea.